src/qibolab/pulses.py

```diff
@@ -363,6 +363,67 @@ def __repr__(self):
         return f"{self.name}({format(self.rel_sigma, '.6f').rstrip('0').rstrip('.')})"
 
 
+class GaussianSquare(PulseShape):
+    r"""GaussianSquare pulse shape.
+
+    Args:
+        rel_sigma (float): relative sigma so that the pulse standard deviation (sigma) = duration / rel_sigma
+        width (float): Percentage of the pulse that is flat
+
+    .. math::
+
+        A\exp^{-\frac{1}{2}\frac{(t-\mu)^2}{\sigma^2}}[Rise] + Flat + A\exp^{-\frac{1}{2}\frac{(t-\mu)^2}{\sigma^2}}[Decay]
+    """
+
+    def __init__(self, rel_sigma: float, width: float):
+        self.name = "GaussianSquare"
+        self.pulse: Pulse = None
+        self.rel_sigma: float = float(rel_sigma)
+        self.width: float = float(width)
+
+    def __eq__(self, item) -> bool:
+        """Overloads == operator."""
+        if super().__eq__(item):
+            return self.rel_sigma == item.rel_sigma and self.width == item.width
+        return False
+
+    def envelope_waveform_i(self, sampling_rate=SAMPLING_RATE) -> Waveform:
+        """The envelope waveform of the i component of the pulse."""
+
+        if self.pulse:
+            num_samples = int(np.rint(self.pulse.duration * sampling_rate))
+            flat_samples = int(np.rint(self.width * num_samples))
+            gaussian_samples = (num_samples - flat_samples) // 2
```

Comment:
The roundings that happen here may cause a missing sample depending on the values of duration, width and sampling rate. For example in
```py
from qibolab.pulses import Pulse, GaussianSquare

pulse = Pulse(start=0, duration=50, amplitude=0.1, frequency=5e9, relative_phase=0, shape=GaussianSquare(5, 0.9))

print(len(pulse.envelope_waveform_i(sampling_rate=1)))
print(len(pulse.envelope_waveform_q(sampling_rate=1)))
```
we expect waveforms of length 50, but we get 49 for the I component and 50 for Q. Similar thing for duration=48 or 49.

It would be good if we can resolve this, because it may cause errors in some drivers, for example if I and Q lengths disagree, or if they are different from the expected length. Putting the missing sample in the flat part is an acceptable solution.

Reply:
This should solve it
```suggestion
            num_samples = int(np.rint(self.pulse.duration * sampling_rate))
            flat_samples = int(np.rint(self.width * num_samples))
            gaussian_samples = (num_samples - flat_samples) // 2
            flat_samples += (num_samples - flat_samples) % 2
```